descartes_planner/src/dense_planner.cpp

```diff
@@ -31,10 +31,10 @@ DensePlanner::~DensePlanner()
 
 }
 
-bool DensePlanner::initialize(descartes_core::RobotModelConstPtr &model)
+bool DensePlanner::initialize(descartes_core::RobotModelConstPtr model)
 {
   planning_graph_ = boost::shared_ptr<descartes_planner::PlanningGraph>(
-      new descartes_planner::PlanningGraph(model));
+      new descartes_planner::PlanningGraph(std::move(model)));
```

Comment:
Would this not prevent incrementing the reference count of the shared pointer? As far as I understand the use of move semantics in this case transfers ownership of the object, so if the containing object goes out of scope could that invalidate other shared pointers?